AnimationPlayground.playground/Contents.swift

```diff
@@ -0,0 +1,437 @@
+        return shape;
+    }
+    class func wmf_addCircle(circle: WMF_Welcome_Circle, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_circlePathWithRadius(toView.frame.size.width * circle.radius).CGPath
+        shape.fillColor = UIColor.blackColor().CGColor
+        shape.strokeColor = UIColor.clearColor().CGColor
+        shape.transform = circle.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: circle.position)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addPlus(plus: WMF_Welcome_Plus, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_plusPathWithWidth(toView.frame.size.width * 0.04).CGPath
+        shape.strokeColor = UIColor.blackColor().CGColor
+        shape.transform = plus.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: plus.position)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addBar(bar: WMF_Welcome_Bar, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        let rect = CGRectMake(
+            0.0,
+            0.0,
+            toView.frame.size.width * bar.frame.size.width,
+            toView.frame.size.height * bar.frame.size.height
+        )
+        shape.path = UIBezierPath(roundedRect: rect, cornerRadius: bar.cornerRadius).CGPath
+        shape.fillColor = bar.color.CGColor
+        shape.transform = bar.initialTransform
+        shape.position = CGPointMake(
+            (toView.frame.size.width * bar.frame.origin.x),
+            (toView.frame.size.height * bar.frame.origin.y)
+        )
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+}
+
+extension CABasicAnimation{
+    class func wmf_animationToXF(transform: CATransform3D, delay: Double, duration: Double) -> CABasicAnimation {
+        let anim = CABasicAnimation(keyPath: "transform")
+        anim.duration = duration
+        anim.beginTime = CACurrentMediaTime() + delay
+        anim.fillMode = kCAFillModeForwards
+        anim.removedOnCompletion = false
+        anim.toValue = NSValue(CATransform3D: transform)
+        anim.timingFunction = CAMediaTimingFunction(name: kCAMediaTimingFunctionEaseInEaseOut)
+        return anim;
+    }
+    class func wmf_animationToOpacity(opacity: Double, delay: Double, duration: Double) -> CABasicAnimation {
+        let anim = CABasicAnimation(keyPath: "opacity")
+        anim.duration = duration
+        anim.beginTime = CACurrentMediaTime() + delay
+        anim.fillMode = kCAFillModeForwards
+        anim.removedOnCompletion = false
+        anim.toValue = opacity
+        anim.timingFunction = CAMediaTimingFunction(name: kCAMediaTimingFunctionEaseInEaseOut)
+        return anim;
+    }
+}
+
+extension UIView{
+    func wmf_addCircles(circles: WMF_Welcome_Circle...) {
```

Comment:
anyway, not sure if we have to time to DRY up all of this, but would be a fun project for you to work on